P0009/P0009.md

```diff
@@ -889,6 +891,52 @@ might only be known at run time.
 This would make a rank-3 array with two compile-time extents
 a natural data structure for the matrices.
 
+Different extents will result in different `mdspan` instantiations,
+even if the element type, rank, layout, and accessor are all the same.
+Some users may be concerned that this will cause a proliferation of types.
+Users with this concern can restrict themselves to only dynamic extents
+and the default layout and accessor.
+They can also "type erase" all three layouts we propose here
+to a single layout type, `layout_stride`.
+
+Our view is that forcing too much type erasure
+(e.g., only permitting dynamic extents,
+or making layout or accessor run-time polymorphic) would hinder performance.
+C++ has generally been moving towards providing more information at compile time.
+Our experience is that compilers optimize array access better
+when given more information about strides at compile time.
+
+We made `mdspan`'s element type, extents, layout, and accessor orthogonal
+in order to simplify customization.
+Even if changing `mdspan`'s design to mix these policies together
+would reduce the number of instantiations,
+we would not do it, because it would make customization harder.
+
+### Conversions between run-time and compile-time extents
+
+We permit converting between `mdspan` types
+that have the same rank, layout, and accessor,
+as long as the extents match at run time.
+This includes both conversion of static extents to dynamic extents,
+and conversion of dynamic extents to static extents.
+Both of these cases are important for helping users made trade-offs
+between compile times and sizes, and run-time performance.
+Static-to-dynamic conversion is a kind of type erasure,
+for when an implementation does not specialize for compile-time extents.
+Dynamic-to-static conversion is a kind of specialization,
+for when an implementation is willing to pay the compile-time cost of specializing.
+One use case for dynamic-to-static conversion is in applications
+that solve partial differential equations through a variety of discretization methods.
+Many methods result in arrays with known fixed extents,
+but making all the extents compile-time constants might cost too much compilation time.
+A rational approach is to make just the most commonly occurring extents compile-time constants.
+
+Converting extents from dynamic to static can cause precondition violations.
```

Comment:
As in std::span converting from dynamic to static can cause precondition violations.